src/dart_document_symbol_provider.ts

```diff
@@ -0,0 +1,62 @@
+"use strict";
+
+import { TextDocument, DocumentSymbolProvider, SymbolInformation, CancellationToken, SymbolKind, Location, Uri, Range, Position } from "vscode";
+import { Analyzer, getSymbolKindForElementKind } from "./analyzer";
+import { toRange } from "./utils";
+import * as as from "./analysis_server_types";
+
+export class DartDocumentSymbolProvider implements DocumentSymbolProvider {
+	private analyzer: Analyzer;
+
+	constructor(analyzer: Analyzer) {
+		this.analyzer = analyzer;
+	}
+
+	provideDocumentSymbols(document: TextDocument, token: CancellationToken): Thenable<SymbolInformation[]> {
+		let file = document.fileName;
+
+		this.analyzer.analysisSetSubscriptions({
+			subscriptions: { 'OUTLINE': [file] }
+		});
+
+		return new Promise<SymbolInformation[]>((resolve, reject) => {
+			let disposable = this.analyzer.registerForAnalysisOutline(n => {
+				if (n.file != file)
+					return;
+
+				this.analyzer.analysisSetSubscriptions({
+					subscriptions: { 'OUTLINE': [] }
+				});
+				disposable.dispose();
+
+				let symbols: SymbolInformation[] = [];
+				for (let element of n.outline.children)
+					this.transcribeOutline(symbols, null, element);
+				resolve(symbols);
+			});
+		});
+	}
+
+	private transcribeOutline(symbols: SymbolInformation[], parent: as.Element, outline: as.Outline) {
+		let element = outline.element;
+
+		let name = element.name;
+		if (element.parameters)
+			name = `${name}()`;
```

Comment:
And the other suggestion:

![Elipsis](https://cloud.githubusercontent.com/assets/1078012/17526338/bcb135e4-5e5e-11e6-9225-5218ae6ddf7b.png)